buildenv/jenkins/jobs/infrastructure/Cleanup-Nodes

```diff
@@ -281,13 +297,33 @@ def sanitize_node(nodeName) {
         println(e.getMessage())
     }
 
+    //reconnect slave
     timeout(time: RECONNECT_TIMEOUT.toInteger(), unit: 'MINUTES') {
-        println("\t ${nodeName}: Reconnecting...")
+        println("\t ${nodeName}: Disconnecting...")
+        waitUntil {
+            // wait for Jenkins to disconnect the slave after kill -9 or reboot
+            return (workingComputer.getNode().getRootPath() == null)
+        }
         workingComputer.disconnect(null)
         workingComputer.waitUntilOffline()
-        workingComputer.connect(false)
-        workingComputer.setTemporarilyOffline(false, null)
-        workingComputer.waitUntilOnline()
-        println("${nodeName} is back online: ${workingComputer.isOnline()}")
+
+        println("\t ${nodeName}: Connecting...")
+        def reconnected = false
+        retry(2) {
```

Comment:
Under what condition would we hit the second retry? I'm not sure if `reconnected` is a good name as it appears to be only used for "If this is the second try, then we should wait 30 seconds".

Reply:
intermittent connection issues, jenkins bugs, I don't expect to happen often, might be overkill

Reply:
I'm torn on adding it as insurance or removing it as unnecessary code.

Reply:
I added a reconnect in the `finally` clause, we can dispense of retry